### ISSUE TITLE
Reproducing results in Caffe.

### ISSUE THREAD

**Issue by mrgloom** [desimone/pytorch-cat-vs-dogs#2]
Hello, I'm trying to reproduce your results in Caffe, I have successfully trained ResNet-101 by finetuning and on 80%-20% train-test split I get accuracy 99.4%, howewer in terms of LogLoss I can't get such good score as yours(my best score is ~0.14).

As I see [here][1] you use `CrossEntropyLoss` and I'm using [SoftmaxWithLoss][2], seems they are the same, so maybe some other tricks were used?

https://github.com/mrgloom/kaggle-dogs-vs-cats-solution

  [1]: https://github.com/desimone/pytorch-cat-vs-dogs/blob/master/main.py#L139
  [2]: http://caffe.berkeleyvision.org/doxygen/classcaffe_1_1SoftmaxWithLossLayer.html

**Comment by desimone** [comment]
SoftmaxWithLoss looks right. `BCELoss` would have worked too. Are you able to reproduce my results using my code? What score are you getting?

I've never used Caffe, so I probably am not the best person to ask, but here are some thoughts.

- Are you using data [augmentation](https://github.com/desimone/pytorch-cat-vs-dogs/blob/master/main.py#L97-L101)? 
- The dataset itself is flawed. I noticed that some of the GT is mislabeled so the randomness of the split could account for a degree of error. 
- Are you [clipping](https://github.com/desimone/pytorch-cat-vs-dogs/blob/master/main.py#L271) your results? Kaggle severely penalizes "1,0" results, so I clipped at a certain threshold.

**Comment by mrgloom** [comment]
Clipping do the trick, using ResNet-101 trained on all data score improvement was from `0.16657` to `0.04971`.

I wonder maybe we can directly optimize for LogLoss? or maybe we can use some nonlinear transform to obtain modified probabilities from original probabilities?

Also I have related question: what is the best practice to obtain probability from SVM classifier? (I'm trying to use CNN as feature extractor and something like Linear SVM for learning on top of them)

Here is some related info:
http://stackoverflow.com/questions/26478000/converting-linearsvcs-decision-function-to-probabilities-scikit-learn-python
http://stackoverflow.com/questions/15111408/how-does-sklearn-svm-svcs-function-predict-proba-work-internally

I have tried CalibratedClassifierCV approach and also using linear_model.LogisticRegression() , accuracy of SVM classifier is about 96% but seems probabilities 'not tuned properly', so log loss highly penalize them.
